src/components/ExtractorResponseEditor/ExtractorResponseEditor.tsx

```diff
@@ -322,10 +325,15 @@ class ExtractorResponseEditor extends React.Component<Props, State> {
         if (pickerProps) {
             this.setState({
                 isSelectionOverlappingOtherEntities: pickerProps.isOverlappingOtherEntities,
-                isMenuVisible: pickerProps.isVisible,
                 menuPosition: pickerProps.position,
                 builtInTypeFilter: pickerProps.builtInTypeFilter
             })
+            if (this.props.isPickerVisible && !pickerProps.isVisible) {
```

Comment:
If both the `isPickerVisible ` and `onClosePicker()` came from props can the parent that set ` isPickerVisible` also call `onClosePicker()` since it laready know that information?  It seems like we're telling a child component to call a function instead of just calling the function.